fe/parameterize_batch.py

```diff
@@ -0,0 +1,523 @@
+            init_conf = init_conf/10 # convert to md_units
+            conf_com = com(init_conf)
+
+            init_mol = Chem.Mol(guest_mol)
+
+            num_conformers = args.num_conformers
+
+            # generate a set of gas phase conformers using the RDKit
+            guest_mol.RemoveAllConformers()
+            AllChem.EmbedMultipleConfs(guest_mol, num_conformers, randomSeed=2020)
+            np.random.seed(2020)
+            for conf_idx in range(num_conformers):
+                conformer = guest_mol.GetConformer(conf_idx)
+                guest_conf = np.array(conformer.GetPositions(), dtype=np.float64)
+                guest_conf = guest_conf/10 # convert to md_units
+                rot_matrix = special_ortho_group.rvs(3).astype(dtype=np.float64)
+                guest_conf = np.matmul(guest_conf, rot_matrix)*10
+
+                for atom_idx, pos in enumerate(guest_conf):
+                    conformer.SetAtomPosition(atom_idx, (float(pos[0]), float(pos[1]), float(pos[2])))
+    
+            host_pdb_file = args.complex_pdb
+            host_pdb = app.PDBFile(host_pdb_file)
+            host_conf = []
+            for x,y,z in host_pdb.positions:
+                host_conf.append([to_md_units(x),to_md_units(y),to_md_units(z)])
+            host_conf = np.array(host_conf)
+            host_name = "complex"
+
+            # set up the system
+            amber_ff = app.ForceField('amber99sbildn.xml', 'amber99_obc.xml')
+            host_system = amber_ff.createSystem(host_pdb.topology,
+                nonbondedMethod=app.NoCutoff,
+                constraints=None,
+                rigidWater=False)
+
+            host_system = openmm_converter.deserialize_system(host_system)
+            num_host_atoms = len(host_system.masses)
+
+            # open_ff = forcefield.Forcefield("ff/smirnoff_1.1.0.py")
+            open_ff = forcefield.Forcefield(args.forcefield)
+            nrg_fns = open_ff.parameterize(guest_mol)
+            guest_masses = get_masses(guest_mol)
+            guest_system = system.System(nrg_fns, open_ff.params, open_ff.param_groups, guest_masses)
+
+            combined_system = host_system.merge(guest_system)
+            cbs = -1*np.ones_like(np.array(combined_system.masses))*0.0001
+            lambda_idxs = np.zeros(len(combined_system.masses), dtype=np.int32)
+            lambda_idxs[num_host_atoms:] = -1
+
+            sim = simulation.Simulation(
+                combined_system,
+                step_sizes,
+                cas,
+                cbs,
+                lambda_schedule,
+                lambda_idxs,
+                precision
+            )
+
+            epoch_train_params = get_params(opt_state)
+
+            epoch_train_ff_params = copy.deepcopy(open_ff)
+            epoch_train_ff_params.params = epoch_train_params[len(host_system.params):]
+            fname = "epoch_"+str(epoch)+"_dG_"+str(true_dG)+"_params"
```

Comment:
why do we put the true_dG in the filename?

Reply:
I use ligand_id just for now, but it won't associate with one particular ligand every epoch since we do shuffle.